### PR TITLE
feat: [sc-15979] Implement documentation.md in api documentation

### PR DESCRIPTION
**Pull request by AStoerkel1**
Story details: https://app.shortcut.com/clarkcan/story/15979
Screenshot 2023-01-04 at 11 13 19 AM
Screenshot 2023-01-04 at 11 13 33 AM
Screenshot 2023-01-04 at 11 13 44 AM

### CODE REVIEW

File: docs/competency-api/documentation.md [hunk 1]
```diff
@@ -5,4 +5,94 @@ nav_order: 1
...
+    Bearer: "Bearer Example.bearer.token"
+}
+```
+
+### Create
+
+#### post:
+```http
+https://api.competency-constructor.clark.center/competencies/:competencyId/documentation
+```
```
Comment (pmanal1) [inline docs/competency-api/documentation.md:25]: Post route missing request body, example, and status codes

Comment (AStoerkel1) [reply docs/competency-api/documentation.md:25]: they are the same for post and patch so I didn't just wanna write them twice, I can make that more obvious

File: docs/competency-api/documentation.md [hunk 2]
```diff
@@ -5,4 +5,94 @@ nav_order: 1
...
-# Documentation
\ No newline at end of file
+# Documentation
+
+## Description
+
+### Headers for all requests
+```json
+{
+    Bearer: "Bearer Example.bearer.token"
```
Comment (Cwagne17) [inline docs/competency-api/documentation.md]: The header is authorization not Bearer. The value of the req.header.authorization is "Bearer eyasdf.asdfa.asdf"

File: docs/competency-api/documentation.md [hunk 3]
```diff
@@ -5,4 +5,94 @@ nav_order: 1
...
+}
+```
+
+### Create
+
+#### post:
```
Comment (Cwagne17) [inline docs/competency-api/documentation.md:22]: Does create not have a body? I would add a description and examples of how to do these.

Comment (Cwagne17) [reply docs/competency-api/documentation.md:22]: Can we add curl request examples of how to make each of these requests also.

File: docs/competency-api/documentation.md [hunk 4]
```diff
@@ -5,4 +5,94 @@ nav_order: 1
...
+| description | false | string | Description of the resource being linked too |
+| uri | false | string | Uri where the resource can be found |
+| userId | true | Object Id: string | The userId matching the Bearer token |
+
+<details closed markdown="block">
+  <summary>
+    Example
+  </summary>
+```json
+{
```
Comment (Cwagne17) [inline docs/competency-api/documentation.md:48]: Can we add a request example?

**Review by Cwagne17 — CHANGES_REQUESTED** [review]

File: docs/competency-api/documentation.md (hunk 3 above)
Comment (AStoerkel1) [reply docs/competency-api/documentation.md:22]: It has the same body as update, I'll make it more clear that they are the same

Comment (JosueMolinaMorales) [reply docs/competency-api/documentation.md:22]: For the curl request part, postman allows you to generate a network request in different languages.

File: docs/competency-api/documentation.md [hunk 5]
```diff
@@ -5,4 +5,94 @@ nav_order: 1
...
+
+#### delete one:
+```http
+https://api.competency-constructor.clark.center/competencies/:competencyId/documentation?ids=:documentationId
+```
+
+#### delete multiple:
+```http
+https://api.competency-constructor.clark.center/competencies/:competencyId/documentation?ids=:documentationId,:documentationId,:documentationId
+```
```
Comment (JosueMolinaMorales) [inline docs/competency-api/documentation.md]: Can you just add short descriptions for these routes?

**Review by JosueMolinaMorales — COMMENTED** [review]
Looks good. After Chris's comments are addressed this will be good.

**Review by JosueMolinaMorales — APPROVED** [review]

**Review by pmanal1 — APPROVED** [review]

File: docs/competency-api/documentation.md [hunk 6]
```diff
@@ -5,4 +5,115 @@ nav_order: 1
...
+
+</details>
+
+### Response codes for Create and Update
+
+| Name | Description |
+|:----:|----|
+| 201(create) 204(update) | Success |
+| 400 | Competency is deprecated, rejected, or published and cannot be updated. Or body is incorrectly formatted. |
```
Comment (mfranz1) [inline docs/competency-api/documentation.md]: Can it be made more obvious that this is a list of 400 errors

Comment (mfranz1) [reply docs/competency-api/documentation.md]: File type could also be unsupported

File: docs/competency-api/documentation.md [hunk 7]
```diff
@@ -5,4 +5,115 @@ nav_order: 1
...
+```http
+https://api.competency-constructor.clark.center/competencies/:competencyId/documentation?ids=:documentationId,:documentationId,:documentationId
+```
+
+### Response
+
+| Name | Description |
+|:----:|----|
+| 201 | Documentation Successfully deleted |
+| 400 | Competency is deprecated, rejected, or published and cannot be updated. Or body is incorrectly formatted. |
```
Comment (mfranz1) [inline docs/competency-api/documentation.md]: A rejected competency should allow documentation to be deleted
Also there may be more errors why delete didn't work. IDK any but I have a feeling there could be

**Review by mfranz1 — CHANGES_REQUESTED** [review]

File: docs/competency-api/documentation.md (hunk 6 above)
Comment (AStoerkel1) [reply docs/competency-api/documentation.md]: Currently this route only accepts text in the body not files

File: docs/competency-api/documentation.md [hunk 8]
```diff
@@ -5,4 +5,118 @@ nav_order: 1
...
+
+{: .warning }
+>Sending no query parameters will delete all documentation associated with this competency
+
+### Query Params
+
+| Name | Required | Type | Description |
+|:----:|:-----:|:----:|-----|
+| ids| false | Object Id: string | Comma separated list of documentation object ids |
+
```
Comment (Cwagne17) [inline docs/competency-api/documentation.md:93]: Can we add examples for the Delete requests?

**Review by Cwagne17 — CHANGES_REQUESTED** [review]

**Review by mfranz1 — APPROVED** [review]

**Review by Cwagne17 — APPROVED** [review]

**Pull request merged**